librz/bin/bin.c

```diff
@@ -1293,3 +1293,149 @@ RZ_API const RzBinLdrPlugin *rz_bin_ldrplugin_get(RzBin *bin, const char *name)
 	}
 	return NULL;
 }
+
+static char *bin_demangle_cxx(RzBinFile *bf, const char *symbol, ut64 vaddr) {
+	char *out = rz_demangler_cxx(symbol);
+	if (!out || !bf) {
+		return out;
+	}
+	char *sign = (char *)strchr(out, '(');
+	if (!sign) {
+		return out;
+	}
+
+	char *str = out;
+	char *ptr = NULL;
+	char *method_name = NULL;
+	for (;;) {
+		ptr = strstr(str, "::");
+		if (!ptr || ptr > sign) {
+			break;
+		}
+		method_name = ptr;
+		str = ptr + 1;
+	}
+
+	if (RZ_STR_ISEMPTY(method_name)) {
+		return out;
+	}
+
+	*method_name = 0;
+	RzBinSymbol *sym = rz_bin_file_add_method(bf, out, method_name + 2, 0);
+	if (sym) {
+		if (sym->vaddr != 0 && sym->vaddr != vaddr) {
+			RZ_LOG_INFO("Duplicated method found: %s\n", sym->name);
+		}
+		if (sym->vaddr == 0) {
+			sym->vaddr = vaddr;
+		}
+	}
+	*method_name = ':';
+	return out;
+}
+
+RZ_API char *bin_demangle_rust(RzBinFile *binfile, const char *symbol, ut64 vaddr) {
+	char *str = NULL;
+	if (!(str = bin_demangle_cxx(binfile, symbol, vaddr))) {
+		return str;
+	}
+	free(str);
+	return rz_demangler_rust(symbol);
+}
+
+RZ_API char *rz_bin_demangle(RzBinFile *bf, const char *language, const char *symbol, ut64 vaddr, bool libs) {
```

Comment:
Please add doxygen, also check `language` for NULL

Reply:
there is a check on language below. it is nullable at the end.